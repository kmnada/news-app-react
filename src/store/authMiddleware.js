import { getCookie } from '../utils/CookieUtil';
import { refreshToken } from './config/api';

/**
 * "If the token is expired, refresh it, then dispatch the action."
 * The first thing we do is check if the token is expired. If it is, we call refreshToken, which is a
 * function that will refresh the token. If the token is not expired, we just dispatch the action
 * @returns A function that takes dispatch as an argument and returns a function that takes next as an
 * argument and returns a function that takes action as an argument and returns a promise.
 */
const AuthMiddleware =
  ({ dispatch }) =>
  (next) =>
  (action) => {
    if (!isTokenExpired()) {
      return refreshToken(dispatch).then((value) => {
        if (value) {
          next(action);
        }
      });
    } else {
      next(action);
    }
  };

export default AuthMiddleware;

/**
 * If the access_token cookie is set, return true, otherwise return false
 */
const isTokenExpired = () => getCookie('access_token');
