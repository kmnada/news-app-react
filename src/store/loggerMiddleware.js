const LoggerMiddleware = (store) => (next) => (action) => {
    console.log("action", action);
    next(action);
  };

export default LoggerMiddleware;