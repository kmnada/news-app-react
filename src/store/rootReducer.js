import { combineReducers } from 'redux';
import loginReducer from './login/reducer';
import storiesReducer from './stories/reducer';

const rootReducer = combineReducers({ storiesReducer, loginReducer });

export default rootReducer;
