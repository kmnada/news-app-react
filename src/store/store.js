import { createStore, applyMiddleware } from 'redux';
import rootReducer from './rootReducer';
import thunkMiddleware from 'redux-thunk';
import LoggerMiddleware from './loggerMiddleware';

const middlewares = [LoggerMiddleware, thunkMiddleware];

/*
Then second argument, ie. the initial state is optional. Based on the type of arguments,
 the createStore function automatically identifies that the passed agument is a middlewares.
 */

export const store = createStore(rootReducer, applyMiddleware(...middlewares));
