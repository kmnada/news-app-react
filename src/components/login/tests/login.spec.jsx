import React from 'react';
import { shallow } from 'enzyme';
import Login from '../login';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';

describe(' Login component', () => {
  it('renders default component', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
