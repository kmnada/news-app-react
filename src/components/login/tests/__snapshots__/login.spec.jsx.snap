// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Login component renders default component 1`] = `ShallowWrapper {}`;
